enhancements/ingress/customize-error-code-pages.md

```diff
@@ -0,0 +1,121 @@
+---
+title: customize-error-code-pages
+authors:
+  - "@miheer"
+reviewers:
+  - "@Miciah"
+  - "@danehans"
+  - "@knobunc"
+approvers:
+  - "@Miciah"
+  - "@danehans"
+  - "@knobunc"
+creation-date: 2021-03-07 
+last-updated: 2020-03-07
+status: implementable 
+see-also:
+replaces:
+superseded-by:
+---
+# Customize error code paged returned by the haproxy router
+## Release Signoff Checklist
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+## Summary
+There is no supported and reasonable method to customize the HAProxy 2.x error page as deployed in OpenShift.
+Customizing the HAProxy error page per our recommended procedure (documented here: https://access.redhat.com/solutions/3425711).
+does not work with OCP 4. It's also important that the right page as per user requirements is served at the right moment.
+Example - 503 error page shall be returned if no pod is available. 
+But when the URI is wrong/does not exist, we should return 404 or a customized 404 page as per user requirements.
+## Motivation
+In OCP 3.11 customizing error pages is possible via a [documented procedure](https://access.redhat.com/solutions/3425711)
+Customers resist customizing the HAProxy error page per our recommended procedure (documented here: https://access.redhat.com/solutions/3425711)
```

Comment:
> Customers resist customizing the HAProxy error page per our recommended procedure (documented here: https://access.redhat.com/solutions/3425711)

Are you saying that customers in OCP 3.11 resist following the recommended procedure? If so, why? Or are you saying customers on OCP 4.x resist following this procedure for updating the HAProxy error pages (simply because it does not work)?

Reply:
Actually that's a different thing. Customers say 503 page is served even when the page is not found when it shall serve 404. The reason I think that happens is because we provide only 503 in our haproxy template

Reply:
I will make it clear